### PR TITLE
Implement keep-alive.

### PR DESCRIPTION
**Pull request by YoEight**

### CODE REVIEW

File: src/grpc.rs [hunk 1]
```diff
@@ -223,6 +234,16 @@ pub struct ClientSettings {
     pub(crate) throw_on_append_failure: bool,
     #[serde(default)]
     pub(crate) default_user_name: Option<Credentials>,
+    #[serde(
+        default = "default_keep_alive_interval_in_ms",
```
Comment (oskardudycz) [inline src/grpc.rs:238]: **SUGGESTION:** Other clients are not using `in_ms` suffixes. Might be worth align that.

**Review by oskardudycz — CHANGES_REQUESTED** [review]
@YoEight I don't see the requirements about the warning implemented:
> if the keepAliveInterval is smaller than ten seconds and bigger or equal 0 <0, 10), then we client or server should log warning information:
>
>$"Specified KeepAliveInterval of {keepAliveInterval} is less than recommended 10_000 ms"

Also, we're throwing ParseError with a generic error while RFC states:

> if the keepAliveInterval is smaller than -1 then error will be thrown/returned:
> 
> $"Invalid keepAliveInterval {keepAliveInterval}. Please provide a positive integer, or -1 to disable.
> 
> if the keepAliveTimeout is smaller than -1 then error will be thrown/returned:

Other clients applied the above recommendations.
$"Invalid keepAliveTimeout {keepAliveTimeout}. Please provide a positive integer, or -1 to disable.

File: tests/fixtures/connection_string/mockups.toml [hunk 2]
```diff
@@ -256,6 +282,25 @@ preference = "Leader"
...
 tls_verify_cert = false
 throw_on_append_failure = true
+keep_alive_interval = 10_000
+keep_alive_timeout = 10_000
 [[mockups.expected.hosts]]
 host = "host"
 port = 2_113
+
+[[mockups]]
+string = "esdb://localhost?keepAliveInterval=20&keepAliveTimeout=10"
```
Comment (oskardudycz) [inline tests/fixtures/connection_string/mockups.toml:292]: **CONCERN:** Can we add more test scenarios? E.g.
- when values equal to `-1` are provided,
- when values smaller than `-1` are provided,
- plus some permutations with valid and invalid values.

File: src/grpc.rs (hunk 1 above)
Comment (YoEight) [reply src/grpc.rs:238]: but that points to a function that returns the default `10s` value, `default` in this case is for where that default value is. It gives context to what would be a magic number instead.

File: tests/fixtures/connection_string/mockups.toml (hunk 2 above)
Comment (YoEight) [reply tests/fixtures/connection_string/mockups.toml]: Done

File: src/grpc.rs (hunk 1 above)
Comment (YoEight) [reply src/grpc.rs]: Done

**Review by oskardudycz — APPROVED** [review]
Looks good!

**Pull request merged**